.circleci/config.yml

```diff
@@ -0,0 +1,426 @@
+
+            sudo killall -9 apt-get || true
+            sudo apt-get -qq update
+            sudo apt-get install -qq -y -f sbt
+      - run:
+          name: Install AdoptOpenJDK << parameters.jdkVersion >>
+          command: |
+            curl -Ls https://git.io/jabba | bash && . ~/.jabba/jabba.sh
+            jabba install "adopt@~1.<< parameters.jdkVersion >>.0-0" && jabba use "$_" && java -Xmx32m -version
+
+            # Save the current version of jabba. This will be used below to reload
+            # this version for every step in the job.
+            jabba current > ~/.jabba/current
+
+            # Add jabba to the env so it will be loaded for every step and also
+            # use the specified jdk version each time it loads.
+            BASH_ENV=${BASH_ENV:-/tmp/bash.env}
+            echo '[ -s ~/.jabba/jabba.sh ] && . ~/.jabba/jabba.sh' >> "$BASH_ENV"
+            echo 'jabba use $(cat ~/.jabba/current)' >> "$BASH_ENV"
+
+
+  run-code-compilation:
+    description: "Compile code for the specified scalaVersion"
+    parameters:
+      scalaVersion:
+        type: string
+        default: "2.12.8"
+    steps:
+      - run:
+          name: Compile code
+          command: sbt "++<< parameters.scalaVersion >> test:compile"
+
+  run-code-validations:
+    description: "Code validations (format, binary compatibility, whitesource, etc.)"
+    steps:
+      - run:
+          name: Check copyright headers
+          command: sbt +headerCheck +test:headerCheck multi-jvm:headerCheck
+      - run:
+          name: Check binary compatibility
+          command: sbt +mimaReportBinaryIssues
+      - run:
+          name: Check Scala code formatting
+          command: sbt scalafmtCheckAll scalafmtSbtCheck
+      - run:
+          name: Check Java code formatting
+          command: |
+            sbt javafmt test:javafmt multi-jvm:javafmt
+            git diff --exit-code || (
+              echo "[error] ERROR: javafmt check failed, see differences above."
+              echo "[error] To fix, before submitting a pull request, format your"
+              echo "[error] sources using sbt javafmt test:javafmt inside docs folder."
+              false
+            )
+
+  store-test-reports:
+    description: "Store test report artifacts"
+    steps:
+      - run:
+          name: Save Test Results
+          command: |
+            mkdir -p ~/test-results/junit
+            find . -type f -regex ".*/target/.*/TEST.*xml" -exec cp -v {} ~/test-results/junit \;
+          when: always
+      - store_test_results:
+          path: ~/test-results
+      - store_artifacts:
+          path: ~/test-results
+
+  run-tests:
+    description: "Run the tests for scalaVersion"
+    parameters:
+      scalaVersion:
+        type: string
+        default: "2.12.8"
+    steps:
+      - run:
+          name: Run tests
+          command: sbt ";set concurrentRestrictions in Global += Tags.limitAll(1) ; ++<< parameters.scalaVersion >> test"
```

Comment:
Not all the sbt runs are using this configuration for `concurrentRestrictions`. That is a good thing in my opinion since we can more precisely optimize for each job.